tests/components/onewire/test_entity_sysbus.py

```diff
@@ -0,0 +1,92 @@
+"""Tests for 1-Wire temperature sensor (device family 10, 22, 28, 3B, 42) connected on SysBus."""
+from os import path
+from unittest.mock import PropertyMock, patch
+
+from pi1wire import (
+    InvalidCRCException,
+    NotFoundSensorException,
+    UnsupportResponseException,
+)
+
+from homeassistant.components.onewire.const import DEFAULT_SYSBUS_MOUNT_DIR, DOMAIN
+from homeassistant.components.onewire.sensor import OneWireDirect
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.setup import async_setup_component
+from homeassistant.util import slugify
+
+from tests.common import mock_registry
+
+MOCK_DEVICE_ID = "28-111111111111"
+MOCK_DEVICE_NAME = "My DS18B20"
+
+
+async def test_onewiredirect_setup(hass):
+    """Test that sysbus config entry works correctly."""
+    entity_registry = mock_registry(hass)
+    config = {
+        "sensor": {
+            "platform": DOMAIN,
+            "names": {
+                MOCK_DEVICE_ID: MOCK_DEVICE_NAME,
+            },
+        }
+    }
+    with patch(
+        "homeassistant.components.onewire.sensor.Pi1Wire"
+    ) as mock_pi1wire, patch("pi1wire.OneWire") as mock_owsensor:
+        type(mock_owsensor).mac_address = PropertyMock(
+            return_value=MOCK_DEVICE_ID.replace("-", "")
+        )
+        mock_owsensor.get_temperature.return_value = "25.123"
+        mock_pi1wire.return_value.find_all_sensors.return_value = [mock_owsensor]
+        assert await async_setup_component(hass, SENSOR_DOMAIN, config)
+        await hass.async_block_till_done()
+
+        assert len(entity_registry.entities) == 1
+        sensor_id = "sensor." + slugify(MOCK_DEVICE_NAME) + "_temperature"
+        sensor_entity = entity_registry.entities.get(sensor_id)
+        assert sensor_entity is not None
+        assert (
+            sensor_entity.unique_id == f"/sys/bus/w1/devices/{MOCK_DEVICE_ID}/w1_slave"
+        )
+        assert sensor_entity.unit_of_measurement == TEMP_CELSIUS
+
+        state = hass.states.get(sensor_id)
+        assert state.state == "25.1"
+
+
+def test_onewiredirect_entity(hass):
+    """Test that onewiredirect updates correctly."""
+    with patch("pi1wire.OneWire") as owsensor:
+        init_args = [
+            MOCK_DEVICE_NAME,
+            path.join(DEFAULT_SYSBUS_MOUNT_DIR, MOCK_DEVICE_ID, "w1_slave"),
+            "temperature",
+            owsensor.return_value,
+        ]
+    test_sensor = OneWireDirect(*init_args)
```

Comment:
Don't interact directly with any entity or integration details.

See:
https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Reply:
What is the best way to test a subsequent Update request on the sensor?
So that the first attempt returns a value, and the second attempt returns FileNotFoundError.

```python
        mock_owsensor.get_temperature.side_effect = [
            25.123,
            FileNotFoundError,
        ]

        # first attempt
        state = hass.states.get(sensor_id)
        assert state.state == "25.1"

        #Request update from hass
        hass.states.get(sensor_id)

        # second attempt
        state = hass.states.get(sensor_id)
        assert state.state is None
```

Reply:
Fire a time changed event with a datetime that is at least one scan interval in the future.

https://github.com/home-assistant/core/blob/e6a65b8bd3a8b9a6a3ecbeb9f5355272b72a29f6/tests/common.py#L287-L288